### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by bryanhce**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -309,18 +338,19 @@ The following activity diagram summarizes what happens when a user enters an `ed
 
 ### Marking all Rooms as Unclean
 
-* In GuestBook, you can mark all the rooms as unclean. This feature was added to make it easier for the hotel to transit to a new working day, as they would usually have to clean all the rooms when a new day starts.
-#### Implementation
+In GuestBook, you can mark all the rooms as unclean. This feature was added to make it easier for the hotel to transit to a new working day, as they would usually have to clean all the rooms when a new day starts.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should we standardize either the user or you?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -448,6 +479,25 @@ As the users of this application would be hotel mangers of small hotels, there w
...
+|------------------------|---------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **Add**                | `add n/NAME p/PHONE e/EMAIL rm/ROOM dr/DATE_RANGE ng/NUMBER_OF_GUESTS [rq/REQUEST]`                                       | `add n/John Doe p/98765432 e/johnd@example.com rm/05-73 dr/19/05/20 - 24/05/22 ng/3` <br> Adds guest John Doe to GuestBook with his relevant details. |
+| **Bill**               | `bill INDEX b/BILL`                                                                                                       | `bill 2 b/99.99` <br> Increases the bill of guest at second index by 99.99.                                                                           |
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should we include the sign after the b/? or like have another example for deduct?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -720,101 +765,88 @@ testers are expected to do more *exploratory* testing.
...
          dr/13/09/22 - 15/09/23 ng/1 rq/Kill the insect `<br>
          Expected: No guest is added,
-         because the room is already in GuestBook. Error details shown in the status message.
+         because the **room is already in GuestBook**. Error details shown in the status message.
          Status bar remains the same.
 
    5. Test case: `add n/John@y Doe p/98765431 e/johnd@nus.com rm/06-73
                   dr/13/09/22 - 15/09/23 ng/1 rq/Kill the insect `<br>
-      Expected: No guest is added, because the name is invalid. Error details shown in the status message.
+      Expected: No guest is added, because the **name** is invalid. Error details shown in the status message.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: What is a status message btw?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -720,101 +765,88 @@ testers are expected to do more *exploratory* testing.
...
    8. Test case: `add n/Johnny Doe p/98765431 e/johnd@nus.com rm/!06-73
                   dr/13/09/22 - 15/09/23 ng/1 rq/Kill the insect `<br>
-      Expected: No guest is added, because the room is invalid. Error details shown in the status message.
+      Expected: No guest is added, because the **room** is invalid. Error details shown in the status message.
       Status bar remains the same.
 
    9. Test case: `add n/Johnny Doe p/98765431 e/johnd@nus.com rm/06-73
                   dr/13/09/22 - 13/09/23 ng/1 rq/Kill the insect `<br>
-      Expected: No guest is added, because the date range is invalid. Error details shown in the status message.
+      Expected: No guest is added, because the **date range** is invalid. Error details shown in the status message.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: I dont think this dateRange is invalid? maybe replace with 13/09/22 - 10/09/22 or something

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -720,101 +765,88 @@ testers are expected to do more *exploratory* testing.
...
+   1. Test case: `add n/John Doe p/98765432 e/johnd@example.com rm/05-73
                   dr/13/09/22 - 15/09/22 ng/1 rq/Apply for room service `<br>
-      Expected: Guest add successfully
+      Expected: Guest add successfully.
 
    3. Test case: `add n/John Doe p/98765431 e/johnd@nus.com rm/06-73
                   dr/13/09/22 - 15/09/23 ng/1 rq/Kill the insect `<br>
       Expected: No guest is added,
-      because the name is already in GuestBook. Error details shown in the status message.
+      because the **name is already in GuestBook**. Error details shown in the status message.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should we use code blocks for all the mentions of fields here? like `Name`, `Phone`, `Email`, `Room`, `Date Range` and `Number Of Guests`?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -720,101 +765,88 @@ testers are expected to do more *exploratory* testing.
...
       Status bar remains the same.
 
-   5. Other incorrect edit commands to try: `edit`, `edit x`, `edit 1 rc/hi`, `edit ...`<br>
-      Expected: No guest is edited, because the parameters are invalid. Error details shown in the status message.
-      Status bar remains the same.
+### Saving data
+
+1. Dealing with corrupted data files
+
+   1. A possible cause of a corrupted file could be because of duplicate room numbers in the data of the Guest.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should we say name as well as room numbers? Also should we use code blocks here?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Good work! Just some comments to consider!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Yes, fixed.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Will include.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: I think this should be result display. Will update.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Yes fixed.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Agreed.

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Agreed!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -708,113 +754,87 @@ testers are expected to do more *exploratory* testing.
...
-      Status bar remains the same.
+1. Dealing with corrupted data files
 
-   5. Other incorrect edit commands to try: `edit`, `edit x`, `edit 1 rc/hi`, `edit ...`<br>
-      Expected: No guest is edited, because the parameters are invalid. Error details shown in the status message.
-      Status bar remains the same.
+   1. A possible cause of a corrupted file could be because of duplicate `name` and/or `room` in the data of the Guest.
+   2. While GuestBook throws an exception when a user tries to add a guest with duplicate `name` and/or `room`, a developer
+   can circumvent this by changing the data in the `guestbook.json` file directly. 
+   3. Similarly, to fix this corrupted file you could either:
```
Comment (jehousoh) [inline docs/DeveloperGuide.md:838]: Should this be the user as well?

**Review by jehousoh — CHANGES_REQUESTED** [review]
One small comment

**Review by jehousoh — APPROVED** [review]
lgtm

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md:838]: This should remain as "you" as we are referring to the developer and not the hotel managers (the user).

**Pull request merged**